atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/LockWatchValueScopingCacheImpl.java

```diff
@@ -120,10 +120,18 @@ public TransactionScopedCache createTransactionScopedCache(long startTs) {
     }
 
     private synchronized void processCommitUpdate(long startTimestamp) {
-        CommitUpdate commitUpdate = eventCache.getCommitUpdate(startTimestamp);
         Optional<TransactionScopedCache> cache = cacheStore.getCache(StartTimestamp.of(startTimestamp));
+        cache.ifPresent(TransactionScopedCache::finalise);
 
-        commitUpdate.accept(new Visitor<Void>() {
+        Map<CellReference, CacheValue> cachedValues = cache.map(TransactionScopedCache::getValueDigest)
+                .map(ValueDigest::loadedValues)
+                .orElseGet(ImmutableMap::of);
+
+        if (cachedValues.isEmpty()) {
+            return;
+        }
```

Comment:
This shortcut is important, as it allows us to avoid throwing in case there has been a leader election and we are not a serializable transaction (because we can then just not push our values centrally, instead of needing to retry).

Reply:
Where do we throw otherwise, `getCommitUpdate()`? But either way, having a shortcut makes sense

Reply:
Yes